src/Futhark/Analysis/PrimExp.hs

```diff
@@ -171,6 +172,11 @@ isF32 = TPrimExp
 isF64 :: PrimExp v -> TPrimExp Double v
 isF64 = TPrimExp
 
+-- | @True@ if the 'PrimExp' is a 'ValueExp'
+isValueExp :: PrimExp v -> Bool
```

Comment:
What is this used for?  Also, it would be better if it returned `Maybe PrimValue`.